README.md

```diff
@@ -1,72 +1,204 @@
+
+```go
+import (
+    "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2016-03-30/compute"
+    "github.com/Azure/go-autorest/autorest"
+    "github.com/Azure/go-autorest/autorest/to"
+)
+
+c := compute.NewVirtualMachinesClient("<subscription_ID>")
+c.Authorizer = autorest.NewBearerAuthorizer("<adal.OAuthTokenProvider>")
+
+vm, err := c.CreateOrUpdate(
+  "<resource_group_name>",
+  "<vm_name>",
+  compute.VirtualMachine{
+    Location: to.StringPtr("westsus2"),
+    VirtualMachineProperties: &compute.VirtualMachineProperties{
+      HardwareProfile: &compute.HardwareProfile{
+        VMSize: compute.StandardDS1V2,
+      },
+      StorageProfile: &compute.StorageProfile{
+        ImageReference: &compute.ImageReference{
+          Publisher: to.StringPtr("Canonical"),
+          Offer:     to.StringPtr("UbuntuServer"),
+          Sku:       to.StringPtr("16.04-LTS"),
+          Version:   to.StringPtr("latest"),
+        },
+        OsDisk: &compute.OSDisk{
+          Name: to.StringPtr("osDisk223344"),
+          Vhd: &compute.VirtualHardDisk{
+            URI: to.StringPtr("https://<storage_account_name>.blob.core.windows.net/azsamplesgocompute/<vm_name>.vhd")
+          },
+          CreateOption: compute.DiskCreateOptionTypesFromImage,
+        },
+      },
+      OsProfile: &compute.OSProfile{
+        ComputerName:  to.StringPtr("<vm_name>"),
+        AdminUsername: to.StringPtr("<username>"),
+        AdminPassword: to.StringPtr("<password>"),
+        LinuxConfiguration: &compute.LinuxConfiguration{
+          SSH: &compute.SSHConfiguration{
+            PublicKeys: &[]compute.SSHPublicKey{
+              compute.SSHPublicKey{
+                Path:    to.StringPtr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", "<username>")),
+                KeyData: to.StringPtr("<ssh_public_key>"),
+              },
+            },
+          },
+        },
+      },
+      NetworkProfile: &compute.NetworkProfile{
+        NetworkInterfaces: &[]compute.NetworkInterfaceReference{
+          compute.NetworkInterfaceReference{
+            ID: "<your_nic_urn>",
+            NetworkInterfaceReferenceProperties: &compute.NetworkInterfaceReferenceProperties{
+              Primary: to.BoolPtr(true),
+            },
+          },
+        },
+      },
+    },
+  },
+  nil,
+)
+```
+
+For a complete VM example see
+[github.com/Azure-Samples/azure-sdk-for-go-samples/compute][2].
+
+For examples of using more services see
+[Azure-Samples/azure-sdk-for-go-samples][3].
+
+[2]: https://github.com/Azure-Samples/azure-sdk-for-go-samples/tree/master/compute
+[3]: https://github.com/Azure-Samples/azure-sdk-for-go-samples
+
+## Authentication
+
+Most operations require an OAuth token for authentication and authorization.
+Following is how to get one using the
+[adal](https://github.com/Azure/go-autorest/tree/master/autorest/adal)
+package.
+
+If you need a service principal (aka Client, App), run `az ad sp create-for-rbac -n "<app_name>"`.
```

Comment:
Does not say where or how you run this. Should link to the portal, mention Cloud Shell, and also link to the CLI overview page. Can also link to the CLI's "create a service principal" article.